Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5032,6 +5032,7 @@ void WebPageProxy::receivedNavigationActionPolicyDecision(WebProcessProxy& proce
         loadedWebArchive,
         replacedDataStoreForWebArchiveLoad
     ] (Ref<WebProcessProxy>&& processNavigatingTo, SuspendedPageProxy* destinationSuspendedPage, ASCIILiteral reason) mutable {
```

Comment:
What if you just made this lambda take a `Ref<FrameProcess>` instead of a `Ref<WebProcessProxy>`?